src/test/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/pullbased/SolutionMappingsHashTableTest.java

```diff
@@ -0,0 +1,495 @@
+package se.liu.ida.hefquin.engine.queryplan.executable.impl.pullbased;
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.sparql.core.Var;
+import org.apache.jena.sparql.engine.binding.Binding;
+import org.junit.Test;
+import se.liu.ida.hefquin.engine.data.SolutionMapping;
+import se.liu.ida.hefquin.engine.data.impl.SolutionMappingUtils;
+import se.liu.ida.hefquin.engine.datastructures.impl.SolutionMappingsHashTable;
+import se.liu.ida.hefquin.engine.queryplan.executable.impl.GenericIntermediateResultBlockImpl;
+
+import java.util.*;
+
+import static org.junit.Assert.*;
+
+public class SolutionMappingsHashTableTest {
+    @Test
+    public void hashTableWithOneInputVariable() {
+        final Var var1 = Var.alloc("v1");
+        final Var var2 = Var.alloc("v2");
+        final Var var3 = Var.alloc("v3");
+
+        final Node x1 = NodeFactory.createURI("http://example.org/x1");
+        final Node x2 = NodeFactory.createURI("http://example.org/x2");
+        final Node y1 = NodeFactory.createURI("http://example.org/y1");
+        final Node y2 = NodeFactory.createURI("http://example.org/y2");
+        final Node y3 = NodeFactory.createURI("http://example.org/y3");
+        final Node z1 = NodeFactory.createURI("http://example.org/z1");
+        final Node z2 = NodeFactory.createURI("http://example.org/z2");
+        final Node z3 = NodeFactory.createURI("http://example.org/z3");
+
+        final GenericIntermediateResultBlockImpl input1 = new GenericIntermediateResultBlockImpl();
```

Comment:
The code here can be simplified. There is no need for such an input1 object. Instead, simply create the solution mappings directly when calling solMHashTable.add(...) below.